src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Services/Structure/BlockStructureOptionProvider.cs

```diff
@@ -0,0 +1,36 @@
+﻿
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.Options;
+
+#if CODE_STYLE
+using OptionSet = Microsoft.CodeAnalysis.Diagnostics.AnalyzerConfigOptions;
+#endif
+
+namespace Microsoft.CodeAnalysis.Structure
+{
+    internal sealed class BlockStructureOptionProvider
+    {
+        private readonly OptionSet _options;
+
+        public BlockStructureOptionProvider(OptionSet options, bool isMetadataAsSource)
+        {
+            _options = options;
+            IsMetadataAsSource = isMetadataAsSource;
+        }
+
+        public bool IsMetadataAsSource { get; }
+
+        public T GetOption<T>(PerLanguageOption2<T> option, string language)
```

Comment:
i'm curious what other options we need.  should we just have a BlockStructureOptions that contains the bools that we need to look at while processing?